src/routes/webhooks/paddle.ts

```diff
@@ -192,68 +180,93 @@ const planChanged = async (data: SubscriptionUpdatedEvent) => {
 
   return (
     (flags?.subscriptionFlags?.cycle as string) !==
-    extractSubscriptionType(data.data?.items)
+    extractSubscriptionCycle(data?.items)
   );
 };
 
+interface AnalyticsExtra {
+  cycle: string;
+  cost: number;
+  currency: string;
+  payment: string;
+  localCost: number;
+  localCurrency: string;
+  payout: TransactionPayoutTotalsNotification | null;
+}
+
+const getAnalyticsExtra = (
+  data: (
+    | SubscriptionUpdatedEvent
+    | SubscriptionCanceledEvent
+    | TransactionCompletedEvent
+  )['data'],
+): Partial<AnalyticsExtra> => {
```

Comment:
Would prefer each of the `data` types is split in its own function (even if code is duplicated a bit) but let's leave it as is since you did not really touched the logic just copied. If we add more events here we will refactor.

Reply:
Can't agree more on that. I was on the verge of actually doing it. But yeah, I'll try to keep the scope small for now and do more maybe once we get more time 🙏